packages/cf.js/src/client.ts

```diff
@@ -0,0 +1,87 @@
+import * as uuid from "uuid";
+
+import { AppFactory } from "./app-factory";
+import { AppInstance } from "./app-instance";
+import {
+  AppDefinition,
+  MessageDataQuery,
+  NodeMessage,
+  NodeMessageType,
+  NodeProvider,
+  QueryType
+} from "./structs";
+
+export enum ClientEventType {
+  INSTALL = "install",
+  PROPOSE_INSTALL = "proposeInstall",
+  REJECT_INSTALL = "rejectInstall"
+}
+
+export interface ClientEvent {
+  readonly eventType: ClientEventType;
+  readonly data: any; // TODO
+}
+
+const NODE_REQUEST_TIMEOUT = 1500;
+
+export class Client {
```

Comment:
I think this needs a better name... this is the equivalent of an `ethers` Provider right?

Reply:
Yep, it might indeed be a good idea to rename it to Provider

Reply:
OTOH, we already have something in our Counterfactual namespace with "Provider" i.e. NodeProvider. Might be a good idea to stick with Client and prevent confusion